frontend/src/main/scala/bloop/bsp/BloopBspDefinitions.scala

```diff
@@ -0,0 +1,16 @@
+package bloop.bsp
+
+import io.circe._
+import io.circe.derivation._
+import ch.epfl.scala.bsp.Uri
+
+object BloopBspDefinitions {
+  final case class BloopExtraBuildParams(
+      clientClassesRootDir: Option[Uri]
+  )
+
+  object BloopExtraBuildParams {
+    val encoder: RootEncoder[BloopExtraBuildParams] = deriveEncoder
```

Comment:
Do we plan to save that into a file or the plan is just to always send that to the bloop server?

Reply:
This is not intended to be saved. `BloopExtraBuildParams` covers any kind of configuration that a client can pass in to tweak bloop's behaviour. That includes `semanticdbVersion` too --it's just a configuration option passed during the initialization handshake-- so that's why we should add it here.
  
Re your specific PR, here's how I'd make the semanticdb version fit in with the infrastructure in `BloopBspServices`:
  
1. Add `semanticdbVersion` to `BloopExtraBuildParams`.
2. Add an explicit parse step  to obtain the `semanticdbVersion` from a decoded `BloopExtraBuildParams`. You can do that inside `initialize` in `BloopBspServices`, in a similar fashion as we do it with `clientClassesRootDir`. If a semanticdb version is found, then we create a `WorkspaceSettings` instance with it.
  
There's a key difference between what is a workspace setting and what is a BSP configuration enabled by the client and that's why I'm being so adamant on modelling as different entities and instantiating the former based on the latter.

Reply:
Cool, thanks for the explanation. I think we can merge this if everything is in order.